code/cloud9/cloud9_instance_stack.yaml

```diff
@@ -0,0 +1,364 @@
+                  sleep 1
+                  done
+                  if [ $(readlink -f /dev/xvda) = "/dev/xvda" ]
+                  then
+                    sudo growpart /dev/xvda 1
+                    STR=$(cat /etc/os-release)
+                    SUB="VERSION_ID=\"2\""
+                    if [[ "$STR" == *"$SUB"* ]]
+                    then
+                      sudo xfs_growfs -d /
+                    else
+                      sudo resize2fs /dev/xvda1
+                    fi
+                  else
+                    sudo growpart /dev/nvme0n1 1
+                    STR=$(cat /etc/os-release)
+                    SUB="VERSION_ID=\"2\""
+                    if [[ "$STR" == *"$SUB"* ]]
+                    then
+                      sudo xfs_growfs -d /
+                    else
+                      sudo resize2fs /dev/nvme0n1p1
+                    fi
+                  fi
+                - mkdir /home/ec2-user/.aws
+                - echo '[default]' > /home/ec2-user/.aws/config
+                - echo 'output = json' >> /home/ec2-user/.aws/config
+                - chmod 600 /home/ec2-user/.aws/config && chmod 600 /home/ec2-user/.aws/credentials
+                - echo 'PATH=$PATH:/usr/local/bin' >> /home/ec2-user/.bashrc
+                - echo 'export PATH' >> /home/ec2-user/.bashrc
+                - echo '=== CLEANING /home/ec2-user ==='
+                - for f in cloud9; do rm -rf /home/ec2-user/$f; done
+                - chown -R ec2-user:ec2-user /home/ec2-user/
+                - echo '=== PREPARE REBOOT in 1 minute with at ==='
+                - FILE=$(mktemp) && echo $FILE && echo '#!/bin/bash' > $FILE && echo 'reboot -f --verbose' >> $FILE && at now + 1 minute -f $FILE
+                - echo "Bootstrap completed with return code $?"
+
+  C9BootstrapAssociation:
+    Type: AWS::SSM::Association
+    DependsOn: C9OutputBucket
+    Properties:
+      Name: !Ref C9SSMDocument
+      OutputLocation:
+        S3Location:
+          OutputS3BucketName: !Ref C9OutputBucket
+          OutputS3KeyPrefix: bootstrapoutput
+      Targets:
+        - Key: tag:SSMBootstrap
+          Values:
+            - Active
+
+  ################## INSTANCE #####################
+  C9InstanceProfile:
+    Type: AWS::IAM::InstanceProfile
+    Properties:
+      Path: "/"
+      Roles:
+        - Ref: C9Role
+
+  C9Instance:
+    Description: "-"
+    DependsOn: C9BootstrapAssociation
+    Type: AWS::Cloud9::EnvironmentEC2
+    Properties:
+      Description: AWS Cloud9 instance
+      AutomaticStopTimeMinutes:
+        Ref: ShutdownTimeout
+      InstanceType:
+        Ref: C9InstanceType
+      Name:
+        Ref: AWS::StackName
+      OwnerArn:
+        !If [Create3rdPartyResources, !Ref OwnerArn, !Ref "AWS::NoValue"]
+      Tags:
+        - Key: SSMBootstrap
+          Value: Active
+        - Key: Environment
+          Value: AWS
+
+  TemplateBucket:
+    Type: "AWS::S3::Bucket"
+    Properties:
+      BucketName: !Ref BucketName
```

Comment:
Do not define bucket name. This will break UPDATES of the cfn if the bucket exists, please remove...